lib/datadog/tracing/correlation.rb

```diff
@@ -48,7 +48,7 @@ def initialize(
           # Dup and freeze strings so they aren't modified by reference.
           @env = Core::Utils::SafeDup.frozen_dup(env || Datadog.configuration.env)
           @service = Core::Utils::SafeDup.frozen_dup(service || Datadog.configuration.service)
-          @span_id = span_id || 0
+          @span_id = (span_id || 0).to_s
```

Comment:
@TonyCTHsu I don't remember: why are we converting the numbers to strings?

Reply:
@marcotc's question suggests that we should include that in the release notes somewhere. 

Apparently, Ruby integers can natively handle 128-bit numbers, so I agree. It's not clear why we need to convert them to strings.

Reply:
@marcotc @ekump The value returned by those methods are meant to be injected into log entries. 

Hence, returning string instead of integer provides finer control for implementing different encoding.

Reply:
>The value returned by those methods are meant to be injected into log entries

Is that the only supported use for the return value? This is part of the public API, so should we be considering more generic usage? 

Generally speaking, I don't think public API methods should be too opinionated on what is done with their return values (within reason) and we should be returning values as the types that best reflect what they are. This contributes to an API being predictable and intuitive for an engineer to use. If I know `trace_ids` are 128-bit integers, and I see a method called `trace_id` I would assume/expect it to return an integer. 

It should be the responsibility of the calling method to convert it to a string and/or encode it a particular way unless we have a good reason to do otherwise.

Reply:
@ekump we have the method `#to_log_format` which is a helper method you use when you want to inject the correlation into your logs safely, that method is opinionated.

Reply:
Here an interesting bit, OTel uses 128-bit trace ids as well, but it represents them as [16-char strings](https://github.com/open-telemetry/opentelemetry-ruby/blob/ff9a7d3af31ad83ed6e07445a9e1609bec53d779/api/lib/opentelemetry/trace.rb#L24-L32). OTel always serializes the trace id as base 16.